python/shotgun_desktop/startup.py

```diff
@@ -718,36 +455,65 @@ def __launch_app(app, splash, connection, app_bootstrap, server, settings):
             raise UpgradeCoreError(
                 "Running a site configuration without the Template Project requires core v0.16.8 "
                 "or higher.",
-                default_site_config
+                pc_path
             )
 
         # If the configuration on disk is not the site configuration, update it to the site config.
         if not tk.pipeline_configuration.is_site_configuration():
             tk.pipeline_configuration.convert_to_site_config()
 
-    if is_auto_path:
-        updates = tk.get_command("updates")
-        updates.set_logger(logger)
-        updates.execute({})
-
-    if not __toolkit_supports_authentication_module(sgtk):
-        raise UpgradeCoreError(
-            "This version of the Shotgun Desktop only supports core 0.16.4 and higher.",
-            default_site_config
-        )
     # initialize the tk-desktop engine for an empty context
-    splash.set_message("Starting desktop engine.")
+    splash.set_message("Resolving context...")
     app.processEvents()
-
     ctx = tk.context_empty()
+
+    splash.set_message("Launching Engine...")
+    app.processEvents()
+
+    # Now start the engine. This import will import the new Toolkit we've bootstrapped into.
+    import sgtk
     engine = sgtk.platform.start_engine("tk-desktop", tk, ctx)
 
     if not __desktop_engine_supports_authentication_module(engine):
         raise UpgradeEngineError(
             "This version of the Shotgun Desktop only supports tk-desktop engine 2.0.0 and higher.",
-            default_site_config
+            pc_path
         )
 
+    return __post_bootstrap_engine(sgtk, splash, app_bootstrap, server, engine)
+
+
+def __zero_config_bootstrap(app, splash, user, app_bootstrap, server, settings):
+    """
+    Launch into the engine using the new zero config based bootstrap.
+
+    :param app: Application object for event processing.
+    :param splash: Splash dialog to update user on what is currently going on
+    :param user: Current ShotgunUser.
+    :param app_bootstrap: Application bootstrap.
+    :param server: The tk_framework_desktopserver.Server instance.
+    :param settings: The application's settings.
+
+    :returns: The error code to return to the shell.
+    """
+    # The startup is up to date, now it's time to bootstrap Toolkit.
+    import sgtk
+    mgr = sgtk.bootstrap.ToolkitManager(user)
+
+    mgr.base_configuration = os.environ.get(
```

Comment:
please add some comments about SHOTGUN_CONFIG_FALLBACK_DESCRIPTOR - when is it useful and how is it different from `TK_BOOTSTRAP_CONFIG_OVERRIDE`?